lib/ChunklistManifestGenerator.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * Created by elad.benedict on 8/12/2015.
+ */
+
+var m3u8Parser = require('./promise-m3u8');
+var qio = require('q-io/fs');
+var Q = require('Q');
+var path = require('path');
+var _ = require('underscore');
+
+module.exports = function(outputFolderPath, filename, dvrWindowSize){
+
+    var currentManifest;
+
+    var manifestFullPath = path.join(outputFolderPath, filename);
+
+    var readCurrentManifest = function readCurrentManifest(){
+        return m3u8Parser.parseM3U8(manifestFullPath);
+    };
+
+    var writeCurrentManifest = function writeCurrentManifest(){
+        return qio.write(manifestFullPath, currentManifest.toString());
+    };
+
+    // Generate an empty M3U8 or read the existing manifest from disk
+    var init = function init()
+    {
+        var manifestPath = path.join(outputFolderPath, filename);
+        return qio.exists(manifestPath).then(function(result)
```

Comment:
I think you better create another abstraction level in order to change easily between file-system and other caching options such as memcache.